diff --git a/model/FullText.inc.php b/model/FullText.inc.php
index ce15442..8e75439 100644
--- a/model/FullText.inc.php
+++ b/model/FullText.inc.php
@@ -130,6 +130,8 @@ class Zotero_FullText {
 				$results->addSuccessful($i, $obj);
 			}
 			catch (Exception $e) {
+				Z_Core::debug($e->getMessage());
+
 				Zotero_DB::rollback();
 				
 				// If item key given, include that
@@ -224,6 +226,9 @@ class Zotero_FullText {
 	 * @return {Array<String>|null} An array of item keys, or null if no results
 	 */
 	public static function searchInLibrary($libraryID, $searchText) {
+		if (! Z_CONFIG::$ELASTICSEARCH_ENABLED) {
+			return null;
+		}
 		$params = [
 			'index' => self::$elasticsearchType . "_index",
 			'type' => self::$elasticsearchType,
