diff --git a/model/Storage.inc.php b/model/Storage.inc.php
index 581aa0a..497eb11 100644
--- a/model/Storage.inc.php
+++ b/model/Storage.inc.php
@@ -689,7 +689,7 @@ class Zotero_Storage {
 				. "Content-Disposition: form-data; name=\"$key\"\r\n\r\n"
 				. $val . "\r\n";
 		}
-		$prefix .= "--$boundary\r\nContent-Disposition: form-data; name=\"file\"\r\n\r\n";
+		$prefix .= "--$boundary\r\nContent-Disposition: form-data; name=\"file\"; filename=\"" . $info->filename ."\"\r\n\r\n";
 		
 		// Suffix
 		$suffix = "\r\n--$boundary--";
