diff --git a/misc/master.sql b/misc/master.sql
index 55cbb1d..a46f31f 100644
--- a/misc/master.sql
+++ b/misc/master.sql
@@ -199,7 +199,7 @@ CREATE TABLE `settings` (
 
 CREATE TABLE `shardHostReplicas` (
   `shardHostID` tinyint(3) unsigned NOT NULL,
-  `address` varchar(75) NOT NULL,
+  `address` varchar(255) NOT NULL,
   `port` smallint(5) unsigned NOT NULL DEFAULT '3306',
   `secure` int(1) unsigned NOT NULL DEFAULT '0',
   `state` enum('up','down') NOT NULL,
@@ -209,7 +209,7 @@ CREATE TABLE `shardHostReplicas` (
 
 CREATE TABLE `shardHosts` (
   `shardHostID` tinyint(3) unsigned NOT NULL,
-  `address` varchar(15) NOT NULL,
+  `address` varchar(255) NOT NULL,
   `port` smallint(5) unsigned NOT NULL DEFAULT 3306,
   `state` enum('up','readonly','down') NOT NULL,
   PRIMARY KEY (`shardHostID`)
