diff --git a/resource/config.js b/resource/config.js
index 4a480e3a4..8f6ef99ed 100644
--- a/resource/config.js
+++ b/resource/config.js
@@ -2,32 +2,32 @@ var ZOTERO_CONFIG = {
 	GUID: 'zotero@chnm.gmu.edu',
 	ID: 'zotero', // used for db filename, etc.
 	CLIENT_NAME: 'Zotero',
-	DOMAIN_NAME: 'zotero.org',
+	DOMAIN_NAME: 'localhost',
 	PRODUCER: 'Digital Scholar',
 	PRODUCER_URL: 'https://digitalscholar.org',
 	REPOSITORY_URL: 'https://repo.zotero.org/repo/',
-	BASE_URI: 'http://zotero.org/',
-	WWW_BASE_URL: 'https://www.zotero.org/',
-	PROXY_AUTH_URL: 'https://zoteroproxycheck.s3.amazonaws.com/test',
-	API_URL: 'https://api.zotero.org/',
-	STREAMING_URL: 'wss://stream.zotero.org/',
+	BASE_URI: 'http://localhost:8080/',
+	WWW_BASE_URL: 'http://localhost:8080/',
+	PROXY_AUTH_URL: '',
+	API_URL: 'http://localhost:8080/',
+	STREAMING_URL: 'ws://localhost:8081/',
 	SERVICES_URL: 'https://services.zotero.org/',
 	API_VERSION: 3,
 	CONNECTOR_MIN_VERSION: '5.0.39', // show upgrade prompt for requests from below this version
 	PREF_BRANCH: 'extensions.zotero.',
 	BOOKMARKLET_ORIGIN: 'https://www.zotero.org',
-	BOOKMARKLET_URL: 'https://www.zotero.org/bookmarklet/',
-	START_URL: "https://www.zotero.org/start",
-	QUICK_START_URL: "https://www.zotero.org/support/quick_start_guide",
-	PDF_TOOLS_URL: "https://www.zotero.org/download/xpdf/",
-	SUPPORT_URL: "https://www.zotero.org/support/",
-	TROUBLESHOOTING_URL: "https://www.zotero.org/support/getting_help",
+	BOOKMARKLET_URL: 'http://localhost:8080/bookmarklet/',
+	START_URL: "http://localhost:8080/start",
+	QUICK_START_URL: "http://localhost:8080/support/quick_start_guide",
+	PDF_TOOLS_URL: "http://localhost:8080/download/xpdf/",
+	SUPPORT_URL: "http://localhost:8080/support/",
+	TROUBLESHOOTING_URL: "http://localhost:8080/support/getting_help",
 	FEEDBACK_URL: "https://forums.zotero.org/",
-	CONNECTORS_URL: "https://www.zotero.org/download/connectors",
-	CHANGELOG_URL: "https://www.zotero.org/support/changelog",
-	CREDITS_URL: 'https://www.zotero.org/support/credits_and_acknowledgments',
-	LICENSING_URL: 'https://www.zotero.org/support/licensing',
-	GET_INVOLVED_URL: 'https://www.zotero.org/getinvolved',
+	CONNECTORS_URL: "http://localhost:8080/download/connectors",
+	CHANGELOG_URL: "http://localhost:8080/support/changelog",
+	CREDITS_URL: 'http://localhost:8080/support/credits_and_acknowledgments',
+	LICENSING_URL: 'http://localhost:8080/support/licensing',
+	GET_INVOLVED_URL: 'http://localhost:8080/getinvolved',
 	DICTIONARIES_URL: 'https://download.zotero.org/dictionaries/',
 };
 
@@ -35,4 +35,4 @@ if (typeof process === 'object' && process + '' === '[object process]'){
 	module.exports = ZOTERO_CONFIG;
 } else {
 	var EXPORTED_SYMBOLS = ["ZOTERO_CONFIG"];
-}
\ No newline at end of file
+}
